Index: app/Console/Commands/CheckOverdueMaterialAccountingOperation.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Console/Commands/CheckOverdueMaterialAccountingOperation.php b/app/Console/Commands/CheckOverdueMaterialAccountingOperation.php
--- a/app/Console/Commands/CheckOverdueMaterialAccountingOperation.php	(revision 4a06bbad1d81abfb574dd0cd855946b867303cec)
+++ b/app/Console/Commands/CheckOverdueMaterialAccountingOperation.php	(revision 585e4918a16e23b5bf9e7d1f13f311e7cc0be338)
@@ -76,12 +76,12 @@
                         break;
                     case 19:
                         //Руководителю получателя
-                        $projectObjectId = $operation->destination_project_object_id;
+                        $projectObjectId = $operation->source_project_object_id;
                         (new q3wMaterialTransferOperationController)->sendTransferNotificationToResponsibilityUsersOfObject($operation, $notificationText, $projectObjectId);
                         break;
                     case 38:
                         //Руководителю отправителя
-                        $projectObjectId = $operation->source_project_object_id;
+                        $projectObjectId = $operation->destination_project_object_id;
                         (new q3wMaterialTransferOperationController)->sendTransferNotificationToResponsibilityUsersOfObject($operation, $notificationText, $projectObjectId);
                         break;
                 }
Index: app/Http/Controllers/q3wMaterial/operations/q3wMaterialTransferOperationController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/q3wMaterial/operations/q3wMaterialTransferOperationController.php b/app/Http/Controllers/q3wMaterial/operations/q3wMaterialTransferOperationController.php
--- a/app/Http/Controllers/q3wMaterial/operations/q3wMaterialTransferOperationController.php	(revision 4a06bbad1d81abfb574dd0cd855946b867303cec)
+++ b/app/Http/Controllers/q3wMaterial/operations/q3wMaterialTransferOperationController.php	(revision 585e4918a16e23b5bf9e7d1f13f311e7cc0be338)
@@ -325,7 +325,7 @@
                 case 7:
                     $operation->operation_route_stage_id = 8;
                     $operation->save();
-                    $this->sendTransferNotification($operation, 'Перемещение завершено', $operation->source_responsible_user_id, $operation->destination_project_object_id);
+                    $this->sendTransferNotification($operation, 'Перемещение завершено', $operation->destination_responsible_user_id, $operation->destination_project_object_id);
                     break;
                 case 9:
                     $operation->operation_route_stage_id = 43;
@@ -384,13 +384,13 @@
                 case 17:
                     $operation->operation_route_stage_id = 18;
                     $operation->save();
-                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя получателя.', $operation->destination_project_object_id);
+                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя отправителя.', $operation->destination_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
                 case 18:
                     $operation->operation_route_stage_id = 19;
                     $operation->save();
-                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя получателя.', $operation->source_project_object_id);
+                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя отправителя.', $operation->source_project_object_id);
                     break;
                 case 19:
                     if ($cancelled){
@@ -405,19 +405,19 @@
                 case 20:
                     $operation->operation_route_stage_id = 21;
                     $operation->save();
-                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем получателя.', $operation->destination_responsible_user_id, $operation->source_project_object_id);
+                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем отправителя.', $operation->destination_responsible_user_id, $operation->source_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
                 case 21:
                     $operation->operation_route_stage_id = 22;
                     $operation->save();
-                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Перемещение завершено руководителем получателя.', $operation->source_project_object_id);
+                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Перемещение завершено руководителем отправителя.', $operation->source_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
                 case 22:
                     $operation->operation_route_stage_id = 23;
                     $operation->save();
-                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем получателя.', $operation->source_responsible_user_id, $operation->destination_project_object_id);
+                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем отправителя.', $operation->source_responsible_user_id, $operation->destination_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
 
@@ -497,19 +497,19 @@
                 case 35:
                     $operation->operation_route_stage_id = 36;
                     $operation->save();
-                    $this->sendTransferNotification($operation, 'Конфликт поставлен под контроль руководителя отправителя.', $operation->source_responsible_user_id, $operation->destination_project_object_id);
+                    $this->sendTransferNotification($operation, 'Конфликт поставлен под контроль руководителя получателя.', $operation->source_responsible_user_id, $operation->destination_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
                 case 36:
                     $operation->operation_route_stage_id = 37;
                     $operation->save();
-                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя отправителя.', $operation->source_project_object_id);
+                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя получателя.', $operation->source_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
                 case 37:
                     $operation->operation_route_stage_id = 38;
                     $operation->save();
-                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя отправителя.', $operation->destination_project_object_id);
+                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Конфликт поставлен под контроль руководителя получателя.', $operation->destination_project_object_id);
                     break;
                 case 38:
                     if ($cancelled){
@@ -524,19 +524,19 @@
                 case 39:
                     $operation->operation_route_stage_id = 40;
                     $operation->save();
-                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем отправителя.', $operation->source_responsible_user_id, $operation->destination_project_object_id);
+                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем получателя.', $operation->source_responsible_user_id, $operation->destination_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
                 case 40:
                     $operation->operation_route_stage_id = 41;
                     $operation->save();
-                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Перемещение завершено руководителем отправителя.', $operation->destination_project_object_id);
+                    $this->sendTransferNotificationToResponsibilityUsersOfObject($operation, 'Перемещение завершено руководителем получателя.', $operation->destination_project_object_id);
                     $this->moveOperationToNextStage($operation->id, $moveToConflict);
                     break;
                 case 41:
                     $operation->operation_route_stage_id = 42;
                     $operation->save();
-                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем отправителя.', $operation->destination_responsible_user_id, $operation->source_project_object_id);
+                    $this->sendTransferNotification($operation, 'Перемещение завершено руководителем получателя.', $operation->destination_responsible_user_id, $operation->source_project_object_id);
                     break;
 
                 //Отмена заявки по ветке отправителя этап 6
@@ -1090,13 +1090,13 @@
             case 11:
                 return Auth::id() == $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 19:
-                return $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
+                return $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 25:
                 return Auth::id() == $operation->source_responsible_user_id || $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 30:
                 return Auth::id() == $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
             case 38:
-                return $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
+                return $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
             default:
                 return false;
         }
@@ -1110,13 +1110,13 @@
             case 11:
                 return Auth::id() == $operation->source_responsible_user_id || $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 19:
-                return $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
+                return $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 25:
                 return Auth::id() == $operation->destination_responsible_user_id || $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id) || $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
             case 30:
                 return Auth::id() == $operation->destination_responsible_user_id || $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
             case 38:
-                return $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
+                return $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
             default:
                 return false;
         }
@@ -1130,13 +1130,13 @@
             case 11:
                 return Auth::id() == $operation->source_responsible_user_id || $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 19:
-                return $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
+                return $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 25:
                 return Auth::id() == $operation->source_responsible_user_id || $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
             case 30:
                 return Auth::id() == $operation->destination_responsible_user_id || $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
             case 38:
-                return $this->isUserResponsibleForMaterialAccounting($operation->source_project_object_id);
+                return $this->isUserResponsibleForMaterialAccounting($operation->destination_project_object_id);
             default:
                 return false;
         }
Index: app/Models/q3wMaterial/operations/q3wMaterialOperation.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Models/q3wMaterial/operations/q3wMaterialOperation.php b/app/Models/q3wMaterial/operations/q3wMaterialOperation.php
--- a/app/Models/q3wMaterial/operations/q3wMaterialOperation.php	(revision 4a06bbad1d81abfb574dd0cd855946b867303cec)
+++ b/app/Models/q3wMaterial/operations/q3wMaterialOperation.php	(revision 585e4918a16e23b5bf9e7d1f13f311e7cc0be338)
@@ -138,10 +138,10 @@
             case 37:
             case 71:
             case 77:
-                $responsibilityUsers = ObjectResponsibleUser::where('object_id', $this->source_project_object_id)->get()->pluck('user_id');
+                $responsibilityUsers = ObjectResponsibleUser::where('object_id', $this->destination_project_object_id)->get()->pluck('user_id');
                 return User::whereIn('id', $responsibilityUsers)->get()->pluck('full_name')->join(';');
             case 19:
-                $responsibilityUsers = ObjectResponsibleUser::where('object_id', $this->destination_project_object_id)->get()->pluck('user_id');
+                $responsibilityUsers = ObjectResponsibleUser::where('object_id', $this->source_project_object_id)->get()->pluck('user_id');
                 return User::whereIn('id', $responsibilityUsers)->get()->pluck('full_name')->join(';');
             case 79:
                 $permissionId = Permission::where('codename', 'material_accounting_write_off_confirmation')->first()->id;
