Index: app/Http/Controllers/q3wMaterial/q3wMaterialController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/Http/Controllers/q3wMaterial/q3wMaterialController.php b/app/Http/Controllers/q3wMaterial/q3wMaterialController.php
--- a/app/Http/Controllers/q3wMaterial/q3wMaterialController.php	(revision aedddd4ca41e94cf949773653411e7096648a6a5)
+++ b/app/Http/Controllers/q3wMaterial/q3wMaterialController.php	(revision 90a626ef2520810d21b917afc3b85a149423feb6)
@@ -502,9 +502,15 @@
         $materialsList = $this->getMaterialTableQuery($projectObjectId, $options)
             ->get();
 
+        unset($options->skip);
+        unset($options->take);
+        $totalCount = $this->getMaterialTableQuery($projectObjectId, $options)
+            ->get()
+            ->count();
+
         return json_encode(array(
                 "data" => $materialsList,
-                "totalCount" => $materialsList->count()
+                "totalCount" => $totalCount
             ),
             JSON_UNESCAPED_UNICODE | JSON_NUMERIC_CHECK);
     }
@@ -514,8 +520,6 @@
         $options = json_decode($request['filterOptions']);
         $projectObjectId = json_decode($request["projectObjectId"]);
 
-
-
         $materialsList = $this->getMaterialTableQuery($projectObjectId, $options)
             ->get()
             ->toArray();
Index: resources/views/materials/material-table.blade.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/resources/views/materials/material-table.blade.php b/resources/views/materials/material-table.blade.php
--- a/resources/views/materials/material-table.blade.php	(revision aedddd4ca41e94cf949773653411e7096648a6a5)
+++ b/resources/views/materials/material-table.blade.php	(revision 90a626ef2520810d21b917afc3b85a149423feb6)
@@ -122,9 +122,8 @@
 
             let materialsTableDataSource = new DevExpress.data.DataSource({
                 store: new DevExpress.data.CustomStore({
-                    key: "id",
+                    loadMode: "processed",
                     load: function (loadOptions) {
-                        /*loadOptions.filter = getLoadOptionsFilterArray();*/
                         dataSourceLoadOptions = loadOptions;
 
                         return $.getJSON("{{route('materials.table.list')}}",
@@ -172,8 +171,13 @@
                                 groupPanel: {
                                     visible: false
                                 },
+                                scrolling: {
+                                    mode: "virtual",
+                                    rowRenderingMode: "virtual",
+                                },
                                 paging: {
-                                    enabled: false
+                                    enabled: true,
+                                    pageSize: 100
                                 },
                                 columns: [
                                     {
@@ -202,8 +206,6 @@
                                                         top: container.parent().offset().top - 8,
                                                         left: container.parent().offset().left + 8
                                                     })
-
-
                                                 }
 
                                                 $(`<div>${options.text}</div>`)
